templates/webhook.bin.erb

```diff
@@ -264,7 +264,13 @@ class Server < Sinatra::Base
 
     def generate_types(environment)
       begin
-        command = "#{$command_prefix} /opt/puppetlabs/bin/puppet generate types --environment #{environment}"
+<% if @user != 'root' -%>
+        # explicitly set working directories to work for non-root accounts
+        generate_suffix = '--confdir /etc/puppetlabs/puppet --codedir /etc/puppetlabs/code --vardir /opt/puppetlabs/puppet/cache'
```

Comment:
If the user is non-root, it seems possible they could use different paths. Should these be configurable, using those paths as the default?

Reply:
They definitely use different paths, that's what I am trying to fix.

Those defaults is where typical paths are. I guess we can pass them as arguments, but ideally we should determine them from 'settings' hash.

Reply:
Either way seems better than static content that may be incorrect.

Reply:
I agree. What method do you prefer?
- add additional parameters to the class
- add facts providing current configuration

I think former is more flexible, in case webhook gains functionality to execute generate types on a remote master

Reply:
Hrm. I'm wondering if maybe this is a stopgap anyway and should go in the webhook's gem. @dhollinger thoughts?

Reply:
This has not been accounted for in the webhook gem yet. Probably create the stopgap here and then open an issue on the gem repo.

Reply:
@vchepkov sorry for the delay. Since I think you're the primary user of this in the short term, are you happy with this stopgap and then will you open an issue on [webhook](https://github.com/voxpupuli/puppet_webhook)?